torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -90,9 +90,9 @@ c10::optional<at::ScalarType> PromoteIntegralType(
 }
 
 template <typename B>
-at::Tensor DoBinaryOp(const at::Tensor& self, const at::Tensor& other,
-                      const B& bin_op) {
-  at::ScalarType dtype = at::result_type(self, other);
+at::Tensor DoBinaryOp(
+    const at::Tensor& self, const at::Tensor& other, const B& bin_op,
+    c10::optional<at::ScalarType> target_type = c10::nullopt) {
```

Comment:
Without this change, doBinaryOP will set the result type to the `at::result_type(self, other)`. If the self and other are both int, at::result_type(Int32, int32) will be int32 but we want to set it to float32 (c10::get_default_dtype)

Reply:
The alternative is we don't call doBinaryOp for true_divide since we don't need doBinaryOp to set the result type for this op. The downside is we need to reimplement the logic of "Try to get the device from lhs, if fail try rhs"

Reply:
Move that code into a function like:
```C++
std::pair<XLATensor, XLATensor> GetBinaryOperands(const at::Tensor& self, const at::Tensor& other);
```

Do not use `DoBinaryOp()` and use the function above in `DoBinaryOp()` and `true_divide()`.